### PR TITLE
@types/bull Update to version 3.4.8. Closes #22673 #23965 #18911

### LINKED ISSUE #18911: @types/bull: Name for a job is missing in typings in `Queue.add()`
- [x] I tried using the `@types/bull` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [ ] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @bgrieder, @JProgrammer, @marshall007

Also reported here https://github.com/OptimalBits/bull/issues/653

### LINKED ISSUE #22673: Bull Missing Discard Method
If you know how to fix the issue, make a pull request instead.

- [X] I tried using the `@types/bull` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [ ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @bgrieder, @JProgrammer, @marshall007, @weeco

If you do not mention the authors the issue will be ignored.

Im Trying to use discard method of a job how cited here: https://github.com/OptimalBits/bull/blob/master/REFERENCE.md#jobdiscard
but index.d.ts does not have this method.

src/queues/pending-trip-request.ts(52,11): error TS2339: Property 'discard' does not exist on type 'Job'.
npm ERR! code ELIFECYCLE
npm ERR! errno 2

### LINKED ISSUE #23965: is there any way to add the third override for the constructor?
Otherwise, one has to parse the redis URL into parts first. 

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c300e37b1e3a141b5ab3019984f10bb4599e596b/types/bull/index.d.ts#L23

add this override: 
```js
new (queueName: string, url: string, opts?: Bull.QueueOptions): Bull.Queue;
```

### PR DESCRIPTION
**Pull request by danmana**
Closes #22673, closes #23965, closes #18911

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/OptimalBits/bull/commits/master/REFERENCE.md
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/bull/index.d.ts [hunk 1]
```diff
@@ -141,13 +148,30 @@ declare namespace Bull {
...
      * TODO: Add a watchdog to check if the job has finished periodically.
      * since pubsub does not give any guarantees.
      */
     finished(): Promise<any>;
 
+    /**
+     * Moves a job to the `completed` queue. Pulls a job from 'waiting' to 'active'
+     * and returns a tuple containing the next jobs data and id. If no job is in the `waiting` queue, returns null.
+     */
+    moveToCompleted(returnValue?: string, ignoreLock?: boolean): Promise<string[] | null>;
```
Comment (aleccool213) [inline types/bull/index.d.ts]: 👍

File: types/bull/index.d.ts [hunk 2]
```diff
@@ -141,13 +148,30 @@ declare namespace Bull {
...
+
+    /**
+     * Moves a job to the `failed` queue. Pulls a job from 'waiting' to 'active'
+     * and returns a tuple containing the next jobs data and id. If no job is in the `waiting` queue, returns null.
+     */
+    moveToFailed(errorInfo: { message: string; }, ignoreLock?: boolean): Promise<void>;
```
Comment (aleccool213) [inline types/bull/index.d.ts]: you could add the returning tuple to the return type here

Comment (danmana) [reply types/bull/index.d.ts]: Done, changed it to `[any, JobId] | null` as described in docs

Comment (danmana) [reply types/bull/index.d.ts]: Done, changed it to `[any, JobId] | null` as described in docs

**Review by aleccool213 — APPROVED** [review]

**Pull request merged**